### PR TITLE
feat: Get drafts and get published functions [FC-0062]

### PR DESCRIPTION
**Pull request by ChrisChV**
Adds :

-  `get_drafts`: Given a list of publishable entities returns current drafts. Not all entities may have current drafts.
- `get_published functions`: Given a list of publishable entities returns those who have a published version.
    Not all entities may have current published version.

**Supporting information**

- Github issue: https://github.com/openedx/modular-learning/issues/234
- Internal link: https://tasks.opencraft.com/browse/FAL-3921
- Used in: https://github.com/openedx/edx-platform/pull/35734

**Testing instructions**

- Verify that tests cover all the new code
- Follow the testing instructions in: https://github.com/openedx/edx-platform/pull/35734

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ChrisChV!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@axim-engineering`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: openedx_learning/apps/authoring/publishing/api.py
Comment (ormsbee) [inline openedx_learning/apps/authoring/publishing/api.py:510]: I was thinking about this more recently. The `entity_id__in` query is reasonable for what this does, but it does get expensive for larger queries. But I think even more importantly, the fact that we'd be passing back a `QuerySet[Draft]` makes it harder for callers to chain queries.

Maybe something like this instead?

```python
def filter_entities(entities: QuerySet[PublishableEntity], has_draft=None, has_published=None) -> QuerySet[PublishableEntity]:
    if has_draft is not None:
        entities = entities.filter(draft__version__isnull=(not has_draft))
    if has_published is not None:
        entities = entities.filter(published__version__isnull=(not has_published))

    return entities
```

Comment (ormsbee) [reply openedx_learning/apps/authoring/publishing/api.py:510]: With the calling code from https://github.com/openedx/edx-platform/pull/35734/files looking something like:

```python
        draft_num_children = authoring_api.filter_entities(collection.entities, has_draft=True).count()
        published_num_children = authoring_api.filter_entities(collection.entities, has_published=True).count()
```

Comment (ChrisChV) [reply openedx_learning/apps/authoring/publishing/api.py]: Thanks for the suggestion. Updated here: https://github.com/openedx/openedx-learning/pull/257/commits/19f542b6b56b8b7e62827acaecdbf9d919f551f9

**Review by pomegranited — APPROVED** [review]
:+1: So clean @ChrisChV ! Really happy with this implementation :)

- [x] I tested this using the PR test instructions and checking test coverage
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>User-facing strings are extracted for translation</strike> N/A

**Review by ormsbee — APPROVED** [review]

**Comment by ormsbee** [comment]
(I'm going to rewrite the commit message, but I need to step away for a bit. Please don't merge this yet.)

**Pull request merged**
